src/compiler/utilities.ts

```diff
@@ -1628,21 +1630,48 @@ namespace ts {
         "\u0085": "\\u0085"  // nextLine
     };
 
+    let singleQuoteEscapedCharsRegExp = /[\\\'\u0000-\u001f\t\v\f\b\r\n\u2028\u2029\u0085]/g;
+    let singleQuoteEscapedCharsMap: Map<string> = {
+        "\0": "\\0",
+        "\t": "\\t",
+        "\v": "\\v",
+        "\f": "\\f",
+        "\b": "\\b",
+        "\r": "\\r",
+        "\n": "\\n",
+        "\\": "\\\\",
+        "\'": "\\\'",
+        "\u2028": "\\u2028", // lineSeparator
+        "\u2029": "\\u2029", // paragraphSeparator
+        "\u0085": "\\u0085"  // nextLine

```

Comment:
To prevent people from shooting themselves in the foot, use our `extend` helper function.